autometa/taxonomy/ncbi.py

```diff
@@ -362,62 +363,92 @@ def parse_merged(self):
-        level=logger.DEBUG,
-    )
-    parser = argparse.ArgumentParser(description="Autometa NCBI utilities class")
-    parser.add_argument("ncbi", help="</path/to/ncbi/database/directory>")
-    parser.add_argument(
-        "--query-taxid",
-        help="query taxid to test NCBI class functionality",
-        default=1222,
-        type=int,
-    )
-    parser.add_argument(
-        "--verbose", help="add verbosity", action="store_true", default=True
-    )
-    args = parser.parse_args()
-    ncbi = NCBI(args.ncbi, args.verbose)
-    query_taxid_parent = ncbi.parent(taxid=args.query_taxid)
-    logger.info(
-        f"{args.query_taxid} Name: {ncbi.name(taxid=args.query_taxid)}\n"
-        f'{args.query_taxid} Order: {ncbi.name(taxid=args.query_taxid, rank="order")}\n'
-        f'{args.query_taxid} Class: {ncbi.name(taxid=args.query_taxid, rank="class")}\n'
-        f"{args.query_taxid} Rank: {ncbi.rank(taxid=args.query_taxid)}\n"
-        f"{args.query_taxid} Lineage:\n{ncbi.lineage(taxid=args.query_taxid)}\n"
-        f"{args.query_taxid} is_common_ancestor {query_taxid_parent}: "
-        f"{ncbi.is_common_ancestor(parent_taxid=query_taxid_parent, child_taxid=args.query_taxid)}\n"
-    )
+        lineage_a_taxids = {ancestor.get("taxid") for ancestor in self.lineage(taxid_A)}
+        lineage_b_taxids = {ancestor.get("taxid") for ancestor in self.lineage(taxid_B)}
+        common_ancestor = lineage_b_taxids.intersection(lineage_a_taxids)
+        common_ancestor.discard(1)  # This discards root
+        return True if common_ancestor else False
+
+    def convert_taxid_dtype(self, taxid):
+        """
+        Converts the given `taxid` to integer. It also checks if the given `taxid` is a positive integer and
+        if is is present in either nodes.dmp or names.dmp.
+
+        NOTE
+        ----
+        Function assumes that if the `taxid` is present in either nodes.dmp or names.dmp, then it is present in both of them.
+
+        Parameters
+        ----------
+        taxid : int
+            identifer for a taxon in NCBI taxonomy databases - nodes.dmp, names.dmp or merged.dmp
+
+        Returns
+        -------
+        int
+            `taxid` if the `taxid` is a positive integer and present in either nodes.dmp or names.dmp or
+            taxid recovered from merged.dmp
+
+        Raises
+        ------
+        ValueError
+            Provided `taxid` is not a positive integer
+        DatabaseOutOfSyncError
+            NCBI databases nodes.dmp, names.dmp and merged.dmp are out of sync with each other
+        """
+        # Checking taxid instance format
+        # This checks if an integer has been added as str, eg. "562"
+        if isinstance(taxid, str):
+            invalid_chars = set(string.punctuation + string.ascii_letters)
+            invalid_chars.discard(".")
+            if {char for char in taxid if char in invalid_chars} or taxid.count(
+                "."
+            ) > 1:
```

Comment:
The formatting changes are made by "black" i.e. the formater we are using. Some of the formattings is also done while making commits via the pre-commit hooks.

Reply:
You may be able to avoid the awkward formatting:

```python
invalid_chars_present = set(taxid).intersection(invalid_chars)
if invalid_chars_present or taxid.count(".") > 1:
    raise ...
```


```python
if set(taxid).intersection(invalid_chars) or taxid.count(".") > 1:
    raise ...
```